BlamLib/BlamLib.Test/Halo2/Extraction.cs

```diff
@@ -323,14 +324,26 @@ public void Halo2_ExtractTagCache(List<int> DatumIndices, bool Recursive, bool O
 				MapsDir = kMapsDirectoryPc;
 			}
 
+			string mainmenu = "mainmenu.map";
+			string shared = "shared.map";
+			string single_player_shared = "single_player_shared.map";
+
 			Program.Halo2.LoadPc(
 				MapsDir + @"mainmenu.map",
 				MapsDir + @"shared.map",
 				MapsDir + @"single_player_shared.map");
-			Assert.IsNotNull(Program.Halo2.PcMainmenu);
-			Assert.IsNotNull(Program.Halo2.PcShared);
-			Assert.IsNotNull(Program.Halo2.PcCampaign);
 
+			bool a = file_name == mainmenu;
+			bool b = file_name == shared;
+			bool c = file_name == single_player_shared;
+			bool result = (a == b) || (b == c);
+
+			if (result)

```

Comment:
`file_name != "mainmenu.map" && file_name != "shared.map" && file_name != "single_player_shared.map"`

Reply:
should perhaps be more complex since map dependency is more complex.
E.g. single_player_shared can depend on mainmenu iirc.